lib/icon-services.coffee

```diff
@@ -0,0 +1,27 @@
+DefaultFileIcons = require './default-file-icons'
+{Emitter} = require 'event-kit'
+
+defaultServices =
+  'file-icons': new DefaultFileIcons
+  'element-icons': null
+
+class IconServices
```

Comment:
Do you use the flexibility of the dynamic service naming in this class? If not, I think I'd prefer explicit methods for setting the file icons and element icons services just to keep things super clear.

Reply:
I'm pretty sure this class can be axed, actually. I kept it because our current implementation uses it: [`lib/file-icons.coffee`](https://github.com/atom/tree-view/blob/b2b673fba128843ac5614dd0637112a9c4b03ba7/lib/file-icons.coffee). I renamed it `IconServices` to disambiguate. This class isn't necessary to integrate with our package, and we can easily do without it.

Reply:
Oh! Wait, no, scratch that. I forgot the `IconServices` class is being used to "select" the most appropriate icon service that's available. Where `file-icons` is, confusingly enough, the old, pre-2017 service that was too limited for the `file-icons` *package*, and `element-icons` is the service which points to, well, `file-icons`'s service.

Confusing, eh? 😀